test/unit/lib/runner/test-runner/test-report-handler-tests.js

```diff
@@ -117,47 +107,216 @@ describe('lib/runner/test-runner/test-report-handler.js', function() {
     it('should increment testCount once', function() {
       testReportHandler._report.testCount = 2;
 
-      testReportHandler.appendTestReport(sampleTestReport);
+      testReportHandler.createTestReport(3, 'testName');
 
       expect(testReportHandler._report.testCount).to.equal(3);
     });
 
-    it('should push the passed in report onto to testReports array', function() {
-      testReportHandler.appendTestReport(sampleTestReport);
+    it('should increment failedTestCount once', function() {
+      testReportHandler._report.failedTestCount = 1;
 
-      expect(testReportHandler._report.testReports).to.deep.equal([
-        {
-          testName: 'The test that failed',
-          error: {
-            instanceName: 'componentA',
-            actionName: 'actionX',
-            failedStep: 'effects',
-          },
-        },
-      ]);
+      testReportHandler.createTestReport(3, 'testName');
+
+      expect(testReportHandler._report.failedTestCount).to.equal(2);
+    });
+
+    it('should create a base report for the test number passed in', function() {
+      testReportHandler.createTestReport(3, 'testName');
+
+      expect(testReportHandler._report.testReports[3]).to.deep.equal({
+        testName: 'testName',
+        status: 'fail',
+      });
+    });
+  });
+
+  describe('appendTestSrdErr', function() {
+    let testReportHandler;
+    let EventEmitter;
+    let EventEmitterInstance;
+
+    beforeEach(function() {
+      mockery.enable({useCleanCache: true});
+      mockery.registerAllowable('../../../../../lib/runner/test-runner/test-report-handler.js');
+
+      EventEmitter = sinon.stub();
+      EventEmitterInstance = {
+        emit: sinon.stub(),
+      };
+      EventEmitter.returns(EventEmitterInstance);
+
+      mockery.registerMock('events', {EventEmitter});
+
+      testReportHandler = require('../../../../../lib/runner/test-runner/test-report-handler.js');
+    });
+
+    afterEach(function() {
+      mockery.resetCache();
+      mockery.deregisterAll();
+      mockery.disable();
+    });
+
+    describe('if stdErr already contains a value inside the test report', function() {
+      it('should concatenate the passed in stdErr to the existing stdErr', function() {
+        testReportHandler._report.testReports[2] = {
+          stdErr: 'previousStdErr ',
+        };
+        testReportHandler.appendTestStdErr('newStdErr', 2);
+
+        expect(testReportHandler._report.testReports[2].stdErr).to.equal('previousStdErr newStdErr');
+      });
+    });
+
+    describe('if stdErr is undefined for the test report', function() {
+      it('should the reports .stdErr to the passed in stdErr', function() {
```

Comment:
I am having trouble understanding this test description. Is there a clearer way to phrase it?